subprojects/parseq/src/test-utils/java/com/linkedin/parseq/ParSeqUnitTestHelper.java

```diff
@@ -332,8 +333,17 @@ public void onPlanCompleted(PlanContext planContext) {
     }
 
     public void await(Task<?> root, long timeout, TimeUnit unit) throws InterruptedException {
-      CountDownLatch latch = _taskDoneLatch.computeIfAbsent(root, key -> new CountDownLatch(1));
-      latch.await(timeout, unit);
+      CountDownLatch latch = _taskDoneLatch.get(root);
+      
+      // If the latch is null, it means that onPlanCompleted was already called which removed the latch.
```

Comment:
Put some comment saying that this API is supposed to be called together with `setupAwait`. Or you can put some javadoc on`setupAwait` to explain this. Otherwise, it is easy to forget the sequence we are assuming here.

Reply:
added! :)